src/Container/AgFeatureGrid/AgFeatureGrid.jsx

```diff
@@ -0,0 +1,1021 @@
+        lockPosition: true,
+        suppressMenu: true,
+        suppressSorting: true,
+        suppressFilter: true,
+        suppressResize: true,
+        suppressMovable: true
+      });
+    }
+
+    let index = -1;
+
+    Object.keys(props).forEach(key => {
+      if (attributeBlacklist.includes(key)) {
+        return;
+      }
+
+      if (props[key] instanceof OlGeomGeometry) {
+        return;
+      }
+
+      if (columnDefs[key] && columnDefs[key].index !== undefined) {
+        index = columnDefs[key].index;
+      } else {
+        ++index;
+      }
+      columns[index] = {
+        headerName: key,
+        field: key,
+        ...columnDefs[key]
+      };
+    });
+
+    return columns;
+  }
+
+  /**
+   * Returns the table row data from all of the given features.
+   *
+   * @return {Array} The table data.
+   */
+  getRowData = () => {
+    const {
+      features
+    } = this.props;
+
+    let data = [];
+
+    features.forEach(feature => {
+      const properties = feature.getProperties();
+      const filtered = Object.keys(properties)
+        .filter(key => !(properties[key] instanceof OlGeomGeometry))
+        .reduce((obj, key) => {
+          obj[key] = properties[key];
+          return obj;
+        }, {});
+
+      data.push({
+        key: this.props.keyFunction(feature),
+        ...filtered
+      });
+    });
+
+    return data;
+  }
+
+  /**
+   * Returns the correspondig feature for the given table row key.
+   *
+   * @param {Number} key The row key to obtain the feature from.
+   * @return {ol.Feature} The feature candidate.
+   */
+  getFeatureFromRowKey = key => {
+    const {
+      features,
+      keyFunction
+    } = this.props;
+
+    const feature = features.filter(feature => keyFunction(feature) === key);
+
+    return feature[0];
+  }
+
+  /**
+   * Returns the correspondig rowNode for the given feature id.
+   *
+   * @param {Number} id The feature's unique id to obtain the row from.
+   * @return {Object} The row candidate.
+   */
+  getRowFromFeatureId = id => {
+    const {
+      grid
+    } = this.state;
+
+    let rowNode;
+
+    grid.api.forEachNode(node => {
+      if (node.data.key === id) {
+        rowNode = node;
+      }
+    });
+
+    return rowNode;
+  }
+
+  /**
+   * Returns the currently selected row keys.
+   *
+   * @return {Number[]} An array with the selected row keys.
+   */
+  getSelectedRowKeys = () => {
+    const {
+      grid
+    } = this.state;
+
+    const selectedRows = grid.api.getSelectedRows();
+
+    return selectedRows.map(row => row.key);
+  }
+
+  /**
+   * Called on row click and zooms the the corresponding feature's extent.
+   *
+   * @param {Object} row The clicked row.
+   */
+  onRowClick = evt => {
+    const {
+      onRowClick
+    } = this.props;
+
+    const row = evt.data;
+    const feature = this.getFeatureFromRowKey(row.key);
+
+    if (isFunction(onRowClick)) {
+      onRowClick(row, feature);
+    }
+
+    this.zoomToFeatures([feature]);
```

Comment:
I think the zoom should only happen if no `onRowClick` function is passed as we might allways want to zoom …

Reply:
True, I moved it into the else block.